src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -907,6 +911,35 @@ def _infer_schema_for_file_format(
 
         return new_schema, schema_to_cast, read_file_transformations, None
 
+    def _infer_schema_from_user_input(
+        self, user_schema: StructType, format: str
+    ) -> Tuple[List, List, List]:
+        if format.lower() != "json":
+            raise ValueError(
+                f"Currently only support user schema for JSON format, got {format} instead"
+            )
```

Comment:
I feel this check and passing `format` to this function is unnecessary. The function never uses `format` other than raising this error and a similar check is already done below